crates/ra_parser/src/grammar/expressions/atom.rs

```diff
@@ -507,3 +508,17 @@ fn try_block_expr(p: &mut Parser, m: Option<Marker>) -> CompletedMarker {
     block(p);
     m.complete(p, TRY_EXPR)
 }
+
+// test box_expr
+// fn foo() {
+//     let x = box 1i32;
+// }
+fn box_expr(p: &mut Parser, m: Option<Marker>) -> CompletedMarker {
```

Comment:
@matklad oops...did I miss something in this method? box syntax inside "tuple"s cannot be parsed.
```rust
struct Foo(Box<i32>);
let a = Foo(box 1i32);
let b = (box 1i32, box 2i32);
```